ui/app/AppLayouts/WalletV2/components/AddAccountPopup.qml

```diff
@@ -0,0 +1,210 @@
+            anchors.top: keyOrSeedPhraseInput.bottom
+            anchors.topMargin: marginBetweenInputs
+        }
+
+        Row {
+            id: accountNameInputRow
+            anchors.left: parent.left
+            anchors.leftMargin: 8
+            anchors.right: parent.right
+            anchors.rightMargin: 10
+            anchors.top: separator.bottom
+            anchors.topMargin: marginBetweenInputs
+            height: 82
+            spacing: 10
+            Item {
+                implicitWidth: 434
+                height: parent.height
+                Input {
+                    id: accountNameInput
+                    anchors.fill: parent
+                    customHeight: 56
+                    placeholderText: qsTrId("enter-an-account-name...")
+                    label: qsTrId("account-name")
+                    validationError: popup.accountNameValidationError
+                }
+            }
+            Item {
+                //emoji placeholder
+                width: 80
+                height: parent.height
+                StyledText {
+                    id: inputLabel
+                    text: "Emoji"
+                    font.weight: Font.Medium
+                    font.pixelSize: 13
+                    color: Style.current.textColor
+                }
+                Rectangle {
+                    width: parent.width
+                    height: 56
+                    anchors.top: inputLabel.bottom
+                    anchors.topMargin: 7
+                    radius: 10
+                    color: "pink"
+                    opacity: 0.6
+                }
+            }
+        }
+
+        Input {
+            id: passwordInput
+            anchors.left: parent.left
+            anchors.leftMargin: 8
+            anchors.right: parent.right
+            anchors.rightMargin: 10
+            implicitHeight: 82
+            customHeight: 56
+            anchors.top: accountNameInputRow.bottom
+            anchors.topMargin: marginBetweenInputs
+            placeholderText: qsTr("Give the account a secure password")
+            label: qsTr("Password")
+            textField.echoMode: TextInput.Password
+            validationError: popup.passwordValidationError
+        }
+    }
+    footer: StatusButton {
+        anchors.top: parent.top
+        anchors.right: parent.right
+        text: loading ? qsTrId("loading") : qsTrId("add-account")
+        enabled: !loading && (passwordInput.text !== "") && (accountNameInput.text !== "")
+                 && (keyOrSeedPhraseInput.correctWordCount || (keyOrSeedPhraseInput.text !== ""))
+
+        MessageDialog {
+            id: accountError
+            title: qsTr("Adding the account failed")
+            icon: StandardIcon.Critical
+            standardButtons: StandardButton.Ok
+        }
+
+        onClicked : {
+            loading = true;
+            if (!validate() || (seedPhraseInserted && !validateSeed())) {
```

Comment:
This may be the wrong way to look at it, but in my opinion, I try to follow the follow guidelines when working on validation:
1. each child component should have its own `validate` function, and the top-level root component should also have a `validate` function that calls each child's `validate`. 
2. In each child, there should be a `Changed` handler that calls `validate`.
4. Create a property in the root component `property bool isValid: false` that tracks the validation status of the form (gets set whenever `root.validate()` is called)
5. The submit button should only be enabled when `isValid` is true. 

An example of the above can be seen in https://github.com/status-im/status-desktop/pull/2168. We likely don't need `resetValidation` nor a `reset` as per the comment above.

Also, @saledjenic introduced a nice `Utils.validateAndReturnError` function that could also be utilised: https://github.com/status-im/status-desktop/pull/2782.

Reply:
Following the above guidelines should eliminate a need to track whether or not a value was entered  in the button click handler as well as the need for separate validation functions (`validate` and `validateSeed`).

Reply:
well, I followed the same logic as it is in Wallet1. There if you try to add a new account with either seed or private, the validation is done once the user presses "Add account" button and as there is no reference to that part [here](https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet?node-id=49%3A1742274) I kept what was there from before.

Reply:
Yes, there are some popups which exist that don't have the type of validation mentioned, but that's not to say we can't improve it while we have the chance 😄  What do you think?

Reply:
PR updated, please have a look!